### PR TITLE
OCT-141: Access PIM translator in microfrontends

### PR DESCRIPTION
**Pull request by tseho**
**TL;DR:** updates `translator` and `user-context` to be used in `create-react-app` micro-frontends; but `@akeneo-pim-community/shared` now depends on `jquery`, `backbone` and `underscore`.

---

**Context**

For historical context, javascript was not planned with a system to import modules. In the old days, it was common to create big javascript files imported with `<script src="app.js"></script>` in html pages.

3 patterns then emerged to handle modules & dependencies:
- **CommonJS** ~2009
  ```
  var translator = require('oro/translator');

  module.exports = {...};
  ```
- **AMD** (Asynchronous Module Definition) ~2010
  ```
  define(['oro/translator'], function(translator) {
    // ...
  });
  ```
- **ECMAScript 6** modules ~2015
  ```
  import translator from 'oro/translator';

  export {...};
  ```

Today’s PIM frontend is built on top of old foundations, as a consequence, **we use the 3 methods in our codebase** with a complex webpack configuration to make it work.

And to make things worst, **the AMD ecosystem is dying**.

- The tool used for writing javascript tests, **jest**, does not support AMD and will never do ⇒ [https://github.com/facebook/jest/issues/17](https://github.com/facebook/jest/issues/17)
- **create-react-app**, that we use to bootstrap micro-frontends, does not support AMD.
- **requirejs** does not seems to be maintained anymore ⇒ [https://github.com/requirejs/requirejs/tags](https://github.com/requirejs/requirejs/tags)

**The problem**

Some of our core libraries, like `router`, `translator`, `user-context` are still in AMD, are still used in all new screens of the PIM, even the ones developed in React.

To palliate the fact that we can't use those in our Micro-frontends with CRA, a provider has been developed to re-implement and simulate the behavior of the PIM core libraries, see [MicroFrontendDependenciesProvider](https://github.com/akeneo/pim-community-dev/blob/2f5265266c6c54907bed8d289396a66e8a1c0a7b/front-packages/shared/src/microfrontend/MicroFrontendDependenciesProvider.tsx).

**The hack**

And since the re-implementation still requires to know about the translations or the routes, additional hacks were developed to copy PIM files into our Micro-frontends environments, see [package.json](https://github.com/akeneo/pim-community-dev/blob/2f5265266c6c54907bed8d289396a66e8a1c0a7b/front-packages/measurement/package.json#L39-L40):
```
    "translation:build": "cp ../../public/js/translation/en_US.js ./src/translations.json",
    "route:build": "cp ../../public/js/fos_js_routes.json ./src/routes.json",
```

If you update translations ? You cannot just reload the page, you have to kill your CRA, re-build the PIM frontend & relaunch.

**The solution**

Without having to rewrite from scratch, we can move PIM core libraries to `@akeneo-pim-community/shared` and update them to be compatible with `create-react-app`.
The current PR shows how to do it for the first two, `translator` & `user-context`.

**How**

I did it in 5 steps:
- move the legacy AMD files to `@akeneo-pim-community/shared/legacy` https://github.com/akeneo/pim-community-dev/pull/18033/commits/d35d89b302a269b282710946516e53bab131492d
- convert them to CommonJS https://github.com/akeneo/pim-community-dev/pull/18033/commits/dba1bbd37cea346efde61816a8a21d1f8c508c2b
- expose **interfaces** with the correct typescript hints https://github.com/akeneo/pim-community-dev/pull/18033/commits/6eb0fadf23958609ddd2f49c0ac018a3d232d699
- provide a retro-compatibility bridge from the old path to the new files https://github.com/akeneo/pim-community-dev/pull/18033/commits/47b3abe8bed554df9ecfa05c30f1c9b5fe2ebb8c
- update MicroFrontendDependenciesProvider to use the new interfaces instead of the hack https://github.com/akeneo/pim-community-dev/pull/18033/commits/18c96e7de870f3bd246a49f5d14342460f78dda8

**What does it mean for our micro-frontends?**

If you remove the translations prop, `MicroFrontendDependenciesProvider` now autoload the translations and it's working out-of-box.
Screenshot_2022-09-28_13-18-40

:warning: **The caveat**

`translator` and `user-context` still requires `jquery`, `backbone` and `underscore`.
Moving them to `@akeneo-pim-community/shared` means that our `shared` package now depends also on those libraries.

We are left with 2 options:
- we rewrite the `translator` and `user-context` libraries without jquery & co **in this PR**.
- we accept that it's a first step towards a better version and postpone the rewrite to later.

I prefer the 2nd option because it's hard to evaluate the impact if we remove them but feel free to change my mind.

### CODE REVIEW

**Review by RiffVentura — APPROVED** [review]
:exploding_head:

**Review by ValentinMumble — APPROVED** [review]
